metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/api/ReferenceSequenceXmlRetrieverThroughEntrezApi.java

```diff
@@ -58,23 +61,43 @@ public ReferenceSequenceXmlRetrieverThroughEntrezApi(String entrezApiKey) {
     }
 
     private String fetchEntrezId(String accession, String entrezDatabase) {
-        return restTemplate.exchange(
+        ResponseEntity<String> response = restTemplate.exchange(
                 entrezApiIdRetrievalUrl, HttpMethod.GET, null, String.class,
-                entrezDatabase, accession, entrezApiKey).getBody();
+                entrezDatabase, accession, entrezApiKey);
+        return response.getBody();
     }
 
     private String fetchEntrezData(String id, String entrezDatabase) {
-        return restTemplate.exchange(
+        ResponseEntity<String> response = restTemplate.exchange(
                 entrezApiAssemblyRetrievalUrl, HttpMethod.GET, null, String.class,
-                entrezDatabase, id, entrezApiKey).getBody();
+                entrezDatabase, id, entrezApiKey);
+        return response.getBody();
     }
 
     @Retryable(maxAttemptsExpression="#{${entrez.api.attempts}}",
             backoff=@Backoff(delayExpression="#{${entrez.api.delay}}"))
     public String getXml(String accession, String entrezDatabase) {
+        // First we query the appropriate Entrez database with an accession to find out the corresponding internal ID.
+        // For example, if we have AJPT01332946.1 and query the nuccore database, it'll return an ID of 428325741.
         String idXml = fetchEntrezId(accession, entrezDatabase);
-        String id = idXml.substring(idXml.indexOf(ID_START_TAG) + ID_START_TAG_LENGTH, idXml.indexOf(ID_END_TAG));
-        return fetchEntrezData(id, entrezDatabase);
+        String id = idXml.substring(idXml.indexOf(ID_START_TAG) + ID_START_TAG.length(), idXml.indexOf(ID_END_TAG));
+
+        // Now, having an internal ID, we query the corresponding Entrez database to get the necessary data.
+        String dataXml = fetchEntrezData(id, entrezDatabase);
+        boolean isAssembly = entrezDatabase.equals("assembly");
+        String DATA_ID_START_TAG = isAssembly ? ASSEMBLY_ID_START_TAG : ID_START_TAG;
+        String DATA_ID_END_TAG = isAssembly ? ASSEMBLY_ID_END_TAG : ID_END_TAG;
+        String idFromData = dataXml.substring(dataXml.indexOf(DATA_ID_START_TAG) + DATA_ID_START_TAG.length(),
+                dataXml.indexOf(DATA_ID_END_TAG));
+        if (idFromData.isEmpty()) {
+            // This check is here to ensure that the result which Entrez returns is actually meaningful. Sometimes,
+            // rarely and sporadically, Entrez does return either an empty XML, or an XML complaining about about
```

Comment:
```suggestion
            // rarely and sporadically, Entrez does return either an empty XML, or an XML complaining about 
```

Reply:
Thank you, fixed in https://github.com/EBIvariation/metadata-ws/pull/82/commits/bab4b4c5b2fb46fc437a4a54b029cd899052db08